public/gatk-tools-public/src/main/java/org/broadinstitute/gatk/tools/CatVariants.java

```diff
@@ -318,15 +320,22 @@ public static void main(String[] args){
         }
     }
 
-    private static class PositionComparator implements Comparator<Pair<Integer,File>> {
+    private static class PositionComparator implements Comparator<Pair<VariantContext,File>> {
+    	
+    	VariantContextComparator comp;
+    	
+    	public PositionComparator(SAMSequenceDictionary dict){
+    		comp = new VariantContextComparator(dict);
+    	}
 
         @Override
-        public int compare(Pair<Integer,File> p1, Pair<Integer,File> p2) {
-            int startPositionP1 = p1.getFirst();
-            int startPositionP2 = p2.getFirst();
-            if (startPositionP1  == startPositionP2)
-                return 0;
-            return startPositionP1 < startPositionP2 ? -1 : 1 ;
+        public int compare(Pair<VariantContext,File> p1, Pair<VariantContext,File> p2) {
+            VariantContext startPositionP1 = p1.getFirst();
+            VariantContext startPositionP2 = p2.getFirst();
+            return comp.compare(startPositionP1, startPositionP2);
+            //if (startPositionP1  == startPositionP2)
+            //    return 0;
+            //return startPositionP1 < startPositionP2 ? -1 : 1 ;
```

Comment:
Add a test to CatVariantsIntegrationTes for checking the sorting logic/comparator.